File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n = nums.size();
        if(n==0) return 0;
        set<int>s;
        for(int i=0;i<n;i++) {
            s.insert(nums[i]);
        }
        int maxi = INT_MIN;
        for(auto it: s) {
            if(s.find(it-1) == s.end()) {
                int cnt=1;
                int itr = it;
                while(s.find(itr+1)!=s.end()) {
                    itr++;
                    cnt++;
                }
                maxi=max(maxi,cnt);
            }
        }
        return maxi;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length==0) return 0;
        Set<Integer> s = new HashSet<>();
        int longest = Integer.MIN_VALUE;
        for(int n :nums) s.add(n);

        for(int n: s) {
            if(!s.contains(n-1)){
                int start=n;
                int count=1;
                while(s.contains(start+1)){
                    count++;
                    start++;
                }
                longest = Math.max(longest, count);
            }

        }
        return longest;
    }
}